File: 1200 A DP Problem/solution.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0) {
        	String[] equation = sc.next().split("=");
        	int CoefficientOfX = 0, Const = 0, coefficient = 0;
        	boolean plus = true, EmptyCoefficient = true;
        	for(int i = 0; i < equation[0].length(); ++i) {
        		if(Character.isDigit(equation[0].charAt(i))) {
					coefficient = coefficient * 10 + equation[0].charAt(i) - '0';
					EmptyCoefficient = false;
				} else if(equation[0].charAt(i) == '+') {
					if(plus)
						Const += coefficient;
					else {
						Const -= coefficient;
						plus = true;
					}
					coefficient = 0;
					EmptyCoefficient = true;
				} else if(equation[0].charAt(i) == '-') {
					if(plus) {
						Const += coefficient;
						plus = false;
					} else
						Const -= coefficient;
					coefficient = 0;
					EmptyCoefficient = true;
				} else if(equation[0].charAt(i) == 'x') {
					if(EmptyCoefficient) {
						if(plus)
							++CoefficientOfX;
						else
							--CoefficientOfX;
					} else{
						if(plus)
							CoefficientOfX += coefficient;
						else
							CoefficientOfX -= coefficient;
					}
					coefficient = 0;
				}
        	}
        	if(plus)
				Const += coefficient;
			else
				Const -= coefficient;
			plus = true;
			coefficient = 0;
			EmptyCoefficient = true;
			for(int i = 0; i < equation[1].length(); ++i) {
        		if(Character.isDigit(equation[1].charAt(i))) {
					coefficient = coefficient * 10 + equation[1].charAt(i) - '0';
					EmptyCoefficient = false;
				} else if(equation[1].charAt(i) == '+') {
					if(plus)
						Const -= coefficient;
					else {
						Const += coefficient;
						plus = true;
					}
					coefficient = 0;
					EmptyCoefficient = true;
				} else if(equation[1].charAt(i) == '-') {
					if(plus) {
						Const -= coefficient;
						plus = false;
					} else
						Const += coefficient;
					coefficient = 0;
					EmptyCoefficient = true;
				}
				else if(equation[1].charAt(i) == 'x') {
					if(EmptyCoefficient) {
						if(plus)
							--CoefficientOfX;
						else
							++CoefficientOfX;
					} else{
						if(plus)
							CoefficientOfX -= coefficient;
						else
							CoefficientOfX += coefficient;
					}
					coefficient = 0;
				}
        	}
        	if(plus)
				Const -= coefficient;
			else
				Const += coefficient;
			if(CoefficientOfX == 0 && Const == 0)
				System.out.println("IDENTITY");
			else if(CoefficientOfX == 0 && Const != 0)
				System.out.println("IMPOSSIBLE");
			else
				System.out.println((int)Math.floor((double)Const / CoefficientOfX * (-1)));
        }
		sc.close();
	}
}

File: 1200 A DP Problem/solution.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
#include <stdbool.h>

int main() {
    int t;
    scanf("%d", &t);
    while(t--) {
        char *buffer = NULL;
        scanf("%ms", &buffer);
        int i = 0, CoefficientOfX = 0, Const = 0, coefficient = 0;
        bool plus = true, EmptyCoefficient = true;
        while(buffer[i] != '=') {
            if(isdigit(buffer[i])) {
                coefficient = coefficient * 10 + buffer[i] - '0';
                EmptyCoefficient = false;
            } else if(buffer[i] == '+') {
                if(plus)
                    Const += coefficient;
                else {
                    Const -= coefficient;
                    plus = true;
                }
                coefficient = 0;
                EmptyCoefficient = true;
            } else if(buffer[i] == '-') {
                if(plus) {
                    Const += coefficient;
                    plus = false;
                } else
                    Const -= coefficient;
                coefficient = 0;
                EmptyCoefficient = true;
            } else if(buffer[i] == 'x') {
                if(plus)
                    CoefficientOfX += EmptyCoefficient + coefficient;
                else
                    CoefficientOfX -= EmptyCoefficient + coefficient;
                coefficient = 0;
                EmptyCoefficient = true;
            }
            ++i;
        }
        if(plus)
            Const += coefficient;
        else
            Const -= coefficient;
        plus = true;
        coefficient = 0;
        EmptyCoefficient = true;
        ++i;
        int len = strlen(buffer);
        while(i < len) {
            if(isdigit(buffer[i])) {
                coefficient = coefficient * 10 + buffer[i] - '0';
                EmptyCoefficient = false;
            } else if(buffer[i] == '+') {
                if(plus)
                    Const -= coefficient;
                else {
                    Const += coefficient;
                    plus = true;
                }
                coefficient = 0;
                EmptyCoefficient = true;
            } else if(buffer[i] == '-') {
                if(plus) {
                    Const -= coefficient;
                    plus = false;
                } else
                    Const += coefficient;
                coefficient = 0;
                EmptyCoefficient = true;
            } else if(buffer[i] == 'x') {
                if(plus)
                    CoefficientOfX -= EmptyCoefficient + coefficient;
                else
                    CoefficientOfX += EmptyCoefficient + coefficient;
                coefficient = 0;
                EmptyCoefficient = true;
            }
            ++i;
        }
        if(plus)
            Const -= coefficient;
        else
            Const += coefficient;
        if(CoefficientOfX == 0 && Const == 0)
            puts("IDENTITY");
        else if(CoefficientOfX == 0 && Const != 0)
            puts("IMPOSSIBLE");
        else
            printf("%.f\n", floor((double)Const / CoefficientOfX * (-1)));
        free(buffer);
    }
    return 0;
}
